src/components/Process/Create/MainContent/QuestionType.tsx

```diff
@@ -9,9 +26,69 @@ interface SelectOption {
   label: string
 }
 
+type MultichoiceWarningModalProps = {
+  isOpen: boolean
+  onClose: () => void
+  updateQuestionType: (newType: SelectorTypes) => void
+  pendingTypeRef: React.MutableRefObject<SelectorTypes | null>
+}
+
+const MultichoiceWarningModal = ({
+  isOpen,
+  onClose,
+  updateQuestionType,
+  pendingTypeRef,
+}: MultichoiceWarningModalProps) => {
+  const cancel = () => {
+    pendingTypeRef.current = null
+    onClose()
+  }
+
+  const confirm = () => {
+    if (pendingTypeRef.current) {
+      updateQuestionType(pendingTypeRef.current)
+      pendingTypeRef.current = null
+    }
+    onClose()
+  }
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} size='md'>
+      <ModalOverlay />
+      <ModalContent p={5}>
+        <ModalHeader p={0}>
+          <Flex flexDirection='column' gap={3}>
+            <Heading size='sm'>
+              <Trans i18nKey='process.question_type.confirm.title'>Do you want to switch to Multiple choice?</Trans>
+            </Heading>
+            <Box fontSize='sm' color='texts.subtle'>
+              <Trans i18nKey='process.question_type.confirm.subtitle'>
+                Multiple-choice allows only one question. Switching will keep only your first question and discard the
+                rest.
+              </Trans>
+            </Box>
+          </Flex>
+        </ModalHeader>
+        <ModalBody p={0}>
+          <Flex justifyContent='flex-end' mt={4} gap={2}>
+            <Button variant='outline' onClick={cancel}>
+              <Trans i18nKey='process.question_type.confirm.cancel_button'>Cancel</Trans>
+            </Button>
+            <Button colorScheme='black' onClick={confirm}>
+              <Trans i18nKey='process.question_type.confirm.confirm_button'>Confirm</Trans>
```

Comment:
Try to search common translations in the locales files rather than creating keys (so long). In this case, use `common.confirm` and `common.cancel`